packages/core/src/modules/credentials/CredentialServiceOptions.ts

```diff
@@ -1,9 +1,40 @@
 import type { AgentMessage } from '../../agent/AgentMessage'
 import type { ConnectionRecord } from '../connections/repository/ConnectionRecord'
 import type { CredentialFormat, CredentialFormatPayload } from './formats'
+import type { CredentialPreviewAttributeOptions } from './models'
 import type { AutoAcceptCredential } from './models/CredentialAutoAcceptType'
 import type { CredentialExchangeRecord } from './repository/CredentialExchangeRecord'
 
+export type FormatDataMessagePayload<
```

Comment:
I think this type could use some code documentation describing what it does. I am all for complex typing to improve the users API but this would take some time to understand without docs.